scripts/binary-validation.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+./gradlew apiCheck -q
+EXIT_CODE=$?
+if [ $EXIT_CODE -ne 0 ]; then
+  echo "❌ apiCheck failed, running apiDump for you..."
```

Comment:
For someone who doesn't understand what `apiCheck` and `apiDump` is, I believe this statement is confusing. I think it would be a good idea to explain *why* `apiCheck` failed and *why* `apiDump` is being run for you. 

This is a little rough, but here is a suggestion: 

```suggestion
  echo "❌ Looks like some 'public' mode was added or removed from the project. If we're not careful, this can break the SDK for customers.\n\nGenerating new .api files for you to add to your commit. Make sure to review these files in your pull request to make sure your modifications were not on purpose."
```